File: 3 - C++/C++ - Course 2 - Classes/Round 3/Henkilo.cpp
#include "henkilo.h"

Henkilo::Henkilo() :
	mNimi("tuntematon"), mIka(0)
{
	cout << "Henkilo -luokan parametriton rakentaja\n";
}
Henkilo::Henkilo(string aNimi, int aIka) :
	mNimi(aNimi), mIka(aIka)
{
	cout << "Henkilo -luokan 2 parametrinen rakentaja\n";
}
Henkilo::Henkilo(string aNimi, int aIka, Osoite aOsoite)
	: mNimi(aNimi), mIka(aIka), mOsoite(aOsoite)
{
	cout << "Henkilo -luokan 3 parametrinen rakentaja\n";
}
Henkilo::~Henkilo()
{
	cout << "Henkilo -luokan purkaja" << endl;
}
Henkilo::Henkilo(const Henkilo& aHenkilo)
	:mNimi(aHenkilo.mNimi), mIka(aHenkilo.mIka), mOsoite(aHenkilo.mOsoite)
{
	cout << "Henkilo -luokan kopiorakentaja " << endl;
}

void Henkilo::setNimi(string aNimi)
{
	mNimi = aNimi;
}
void Henkilo::setIka(int aIka)
{
	mIka = aIka;
}
void Henkilo::setOsoite(Osoite aOsoite)
{
	mOsoite = aOsoite;
}
Osoite Henkilo::getOsoite() const
{
	return mOsoite;
}
string Henkilo::getNimi() const
{
	return mNimi;
}
int Henkilo::getIka() const
{
	return mIka;
}
void Henkilo::tulostaHenkilonTiedot() const
{
	cout << "Henkilon '" << mNimi << "' tiedot" << endl;
	cout << "Nimi: " << mNimi << endl;
	cout << "Ika: " << mIka << endl;
	mOsoite.tulostaTiedot();
}
void Henkilo::tervehdi() const
{
	cout << "Hei nimeni on " << mNimi << endl;
	cout << "Olen " << mIka << " vuotias." << endl;
	cout << "Osoitteeni on " << endl;
	mOsoite.tulostaTiedot();
}

void Henkilo::kasva()
{
	cout << "\nSyntymapaiva! \n" << endl;
	mIka++;
}

File: 5 - Ohjelmoinnin syventavat tekniikat/Viikko 2 - 4/Henkilo.java
public class Henkilo {
    private String mNimi;
    private int mIka;
    private Osoite mOsoite;

    public Henkilo(String aNimi, int aIka){
        mNimi = aNimi;
        mIka = aIka;
    }
    public Henkilo(String aNimi, int aIka,Osoite aOsoite) {
        mNimi = aNimi;
        mIka = aIka;
        mOsoite = aOsoite;
    }

    public void tulostaTiedot(){

        System.out.println("Nimi: "+mNimi);
        System.out.println("Ika: " + mIka);
        mOsoite.tulostaTiedot();

    }
    public String getNimi() {
        return mNimi;
    }
    public int getIka() {
        return mIka;
    }
    public Osoite getmOsoite() {
        return mOsoite;
    }

    public void setmNimi(String aNimi) {
        mNimi = aNimi;
    }

    public void setIka(int aIka) {
        mIka = aIka;
    }
    public void setmOsoite(Osoite aOsoite) {
        mOsoite = aOsoite;
    }
}
